redisinsight/ui/src/components/item-list/ItemList.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  Criteria,
+  Direction,
+  EuiBasicTableColumn,
+  EuiInMemoryTable,
+  EuiTableSelectionType,
+  PropertySort
+} from '@elastic/eui'
+import cx from 'classnames'
+import { first, last } from 'lodash'
+import React, { useEffect, useRef, useState } from 'react'
+import { BrowserStorageItem } from 'uiSrc/constants'
+import { localStorageService } from 'uiSrc/services'
+import { TelemetryEvent, sendEventTelemetry } from 'uiSrc/telemetry'
+import { Maybe, Nullable } from 'uiSrc/utils'
+
+import { ActionBar, DeleteAction, ExportAction } from './components'
+
+import './styles.scss'
+import styles from './styles.module.scss'
+
+export type TelemetryEventKeys = 'exportClicked' | 'listSorted' | 'multipleDeleteClicked'
+export type BrowserStorageItemKeys = 'sort'
+
+export interface Props<T> {
+  width: number
+  dialogIsOpen: boolean
+  editedInstance: Nullable<T>
+  columnVariations: EuiBasicTableColumn<T>[][]
+  onDelete: (ids: T[]) => void
+  onExport: (ids: T[], withSecrets: boolean) => void
+  onWheel: () => void
+  loading: boolean
+  data: T[]
+  browserStorageItems: { [key in BrowserStorageItemKeys]: BrowserStorageItem }
+  telemetryEvents: { [key in TelemetryEventKeys]: TelemetryEvent }
```

Comment:
you should not work with storage and telemetry events in this component
it's better to use callback functions (onExport, onSortChanged, onDelete and etc.)